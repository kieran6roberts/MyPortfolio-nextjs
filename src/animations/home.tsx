const regVariant = {
    visible: {
      opacity: 1,
      transition: {
        duration: 1
      }
    },
    hidden: {
      opacity: 0
    }
  };

  const staggerVariant = {
    visible: {
      opacity: 1,
      transition: {
        duration: 0.3,
        delayChildren: 0.2,
        staggerChildren: 0.2
      }
    },
    hidden: {
      opacity: 0,
    }
  };

  export {
      regVariant,
      staggerVariant
  };