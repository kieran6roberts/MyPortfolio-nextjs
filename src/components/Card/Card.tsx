import * as React from "react";
import { motion as m } from "framer-motion";

type COMBINABLE = number | string;
type ANIMATION_VARIANTS = {
    hidden: { [key: string]: COMBINABLE },
    visible: { 
        transition: { [key: string]: COMBINABLE },
        [key: string]: any
    },
};

interface CARDS {
    items: string[],
    header: string,
    parentVariant: ANIMATION_VARIANTS,
    childVariant: ANIMATION_VARIANTS
};

export function generateKey(item: string): string {
    return `${item}_${ new Date().getTime() }`;
};
    
export default function Card({ items, header, parentVariant, childVariant }: CARDS): React.ReactElement {
    return (
        <m.article className="flex flex-col pb-4 mx-2 shadow-lg lg:flex-1"
        variants={parentVariant}>
            <m.h3 className="pt-12 pb-4 font-bold text-center uppercase text-md text-sec"
            variants={childVariant}>
                {header}
            </m.h3>
            <m.ul className="flex flex-col items-center justify-start h-full pt-4 text-xs text-gray-600 bg-light"
            variants={parentVariant}>
                {items && items.map(item =>
                    <m.li key={generateKey(item)}
                    className="my-2"
                    variants={childVariant}>
                        {item}
                    </m.li>                    
                )}
            </m.ul>
        </m.article>
    )
};