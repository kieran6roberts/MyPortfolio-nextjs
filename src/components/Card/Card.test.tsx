import * as React from "react";
import TestRenderer from "react-test-renderer";
import { render, RenderResult } from "@testing-library/react";

import Card from "@/components/Card/Card";
import { regVariant, staggerVariant } from "@/animations/home";

let documentBody: RenderResult;

describe("<Button />", () => {
    const props = {
        items: [ "javascript", "css" ],
        header: "my skills",
        childVariant: regVariant,
        parentVariant: staggerVariant
    };

    test("renders", () => {
        documentBody = render(<Card {...props} />);
        
        expect(documentBody.getByRole("heading", { name: "my skills" })).toBeInTheDocument();
        expect(documentBody.getByRole("list")).toContainElement(documentBody.getAllByRole("listitem")[0]);
        expect(documentBody.getByRole("list")).toContainElement(documentBody.getAllByRole("listitem")[1]);
    });

    test("render snapshot", () => {
        const tree = TestRenderer
        .create(<Card {...props} />)
        .toJSON();

        expect(tree).toMatchSnapshot();
    });
});