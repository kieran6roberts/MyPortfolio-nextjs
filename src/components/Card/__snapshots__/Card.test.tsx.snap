// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> render snapshot 1`] = `
<article
  className="flex flex-col pb-4 mx-2 shadow-lg lg:flex-1"
  style={Object {}}
>
  <h3
    className="pt-12 pb-4 font-bold text-center uppercase text-md text-sec"
    style={Object {}}
  >
    my skills
  </h3>
  <ul
    className="flex flex-col items-center justify-start h-full pt-4 text-xs text-gray-600 bg-light"
    style={Object {}}
  >
    <li
      className="my-2"
      style={Object {}}
    >
      javascript
    </li>
    <li
      className="my-2"
      style={Object {}}
    >
      css
    </li>
  </ul>
</article>
`;
